src/test/java/nearsoft/academy/bigdata/recommendation/MovieRecommender.java

```diff
@@ -0,0 +1,175 @@
+            InputStream gzipStream = new GZIPInputStream(fileStream);
+            Reader decoder = new InputStreamReader(gzipStream, "UTF-8");
+            BufferedReader buffered = new BufferedReader(decoder);
+
+            int usersCounter = 1;
+            int productsCounter = 1;
+            String content;
+
+            while ((content = buffered.readLine()) != null){
+                if (content.contains(userKey)){
+                    String userId = content.split(userKey)[1];
+                    users.add(userId);
+                    if(!this.usersMap.containsKey(userId)){
+                        this.usersMap.put(userId, usersCounter);
+                        usersCounter++;
+                    }
+                }
+                else if(content.contains(productKey)){
+                    String productId = content.split(productKey)[1];
+                    products.add(productId);
+                    if(!this.productsMap.containsKey(productId)){
+                        this.productsMap.put(productId, productsCounter);
+                        productsCounter++;
+                    }
+                }
+                else if(content.contains(reviewKey)){
+                    String scoreValue = content.split(reviewKey)[1];
+                    reviews.add(Double.valueOf(scoreValue));
+                }
+            }
+
+            System.out.println("users: " + usersMap.size() + " products: " + productsMap.size() + " reviews: " + reviews.size());
+            this.writeCsv();
+
+        }catch (Exception e){
+            e.printStackTrace();
+        }
+    }
+
+    int getTotalReviews(){
+        return reviews.size();
+    }
+
+    int getTotalProducts(){
+        return productsMap.size();
+    }
+
+    int getTotalUsers(){
+        return usersMap.size();
+    }
+
+
+    private void writeCsv(){
+        FileWriter fileWriter = null;
+        try {
+            fileWriter = new FileWriter("movies.csv");
+
+            for(int i = 0; i < this.users.size()-1; i++){
+                int userId = this.usersMap.get(this.users.get(i));
+                fileWriter.append(String.valueOf(userId));
+                fileWriter.append(",");
+
+                int productId = this.productsMap.get(this.products.get(i));
+                fileWriter.append(String.valueOf(productId));
+                fileWriter.append(",");
+
+                fileWriter.append(String.valueOf(this.reviews.get(i)));
+
+                fileWriter.append("\n");
+            }
+
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        finally {
+            try {
+                assert fileWriter != null;
+                fileWriter.flush();
+                fileWriter.close();
+            }catch (IOException e) {
+                e.printStackTrace();
+            }
+        }
+    }
+
+    List<String> getRecommendationsForUser(String userId) {
+        List<String> recommended = new ArrayList<String>();
```

Comment:
Nice :) also, depending on Java version, you can use the empty diamond in the `ArrayList<String>` declaration :)